inference-engine/src/gna_plugin/transformations/convert_matmul_to_pointwise_convolution.cpp

```diff
@@ -0,0 +1,135 @@
+        auto weights_node = matmul_node->input_value(1).get_node_shared_ptr();
+        std::shared_ptr<ngraph::Node> matmul_output =
+            matmul_node->output(0).get_target_inputs().begin()->get_node()->shared_from_this();
+        std::shared_ptr<ngraph::Node> bias_node = nullptr;
+        if (std::dynamic_pointer_cast<ngraph::opset7::Add>(matmul_output)) {
+            auto add_second_input = matmul_output->input_value(1).get_node_shared_ptr();
+            if (std::dynamic_pointer_cast<ngraph::opset7::Constant>(matmul_output)) {
+                bias_node = matmul_output->input_value(1).get_node_shared_ptr();
+                matmul_output = matmul_output->output(0).get_target_inputs().begin()->get_node()->shared_from_this();
+            }
+        }
+        std::shared_ptr<ngraph::Node> fq_node = std::dynamic_pointer_cast<ngraph::opset7::FakeQuantize>(matmul_output);
+
+        auto reshape_const_before = std::make_shared<ngraph::opset7::Constant>(ngraph::element::Type_t::i64,
+                                                                               ngraph::Shape{4},
+                                                                               ngraph::Shape{1, 1, width, in_channels});
+        auto reshape_before =  std::make_shared<ngraph::opset7::Reshape>(input_node, reshape_const_before, false);
+        reshape_before->set_friendly_name(matmul_node->get_friendly_name() + "/reshape_in");
+
+        auto transpose_before = std::make_shared<ngraph::opset7::Transpose>(reshape_before,
+            ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape{4},
+            GetPermuteOrder(InferenceEngine::Layout::NHWC, InferenceEngine::Layout::NCHW)));
+        transpose_before->set_friendly_name(matmul_node->get_friendly_name() + "/transpose_in");
+
+        auto weights_reshape_const = std::make_shared<ngraph::opset7::Constant>(ngraph::element::Type_t::i64,
+            ngraph::Shape{4}, ngraph::Shape{out_channels, in_channels, 1, 1});
+        auto weights_reshaped =  std::make_shared<ngraph::opset7::Reshape>(weights_node, weights_reshape_const, false);
+
+        std::shared_ptr<ngraph::Node> conv_node = std::make_shared<ngraph::opset7::Convolution>(transpose_before, weights_reshaped,
+                ngraph::Strides{1, 1}, ngraph::CoordinateDiff{0, 0}, ngraph::CoordinateDiff{0, 0},
+                ngraph::Strides{1, 1}, ngraph::op::PadType::VALID);
+        conv_node->set_friendly_name(matmul_node->get_friendly_name() + "/conv");
+
+        if (bias_node) {
+            conv_node = std::make_shared<ngraph::opset7::Add>(conv_node, bias_node);
+        }
+
+        if (fq_node) {
+            conv_node = fq_node->clone_with_new_inputs({conv_node, fq_node->input_value(1), fq_node->input_value(2),
+                fq_node->input_value(3), fq_node->input_value(4)});
+            ngraph::replace_output_update_name(fq_node->output(0), fq_node->input_value(0));
```

Comment:
why are you doing this?
When you will replace matmul `ngraph::replace_node(matmul_node, reshape_after);` this sub-graph will be eliminated.

Reply:
removed